aiocoap/protocol.py

```diff
@@ -689,6 +703,54 @@ def _add_response_properties(response, request):
                 return
 
 
+class MulticastRequest(interfaces.Request):
+    def __init__(self, plumbing_request, loop, log, timeout=MULTICAST_REQUEST_TIMEOUT):
+
+        r = plumbing_request.request
+        if r.mtype != NON or r.code != GET or r.payload:
+            raise ValueError("Multicast currently only has support for NON GET requests")
+
+        if r.opt.observe == 0:
+            raise ValueError("Multicast currently does not support Observe requests")
+
+        self._plumbing_request = plumbing_request
+        self._timeout_value = timeout
+        self._loop = loop
+
+        self.responses = MulticastResponse()
+
+        loop.create_task(self._run())
+
+        self.log = log
+
+    @staticmethod
+    def _add_response_properties(response, request):
+        response.request = request
+
+    def _timeout(self):
+        self._plumbing_request.stop_interest()
+        self.responses.error(error.WaitingForClientTimedOut())
+        self.responses.cancel()
+        self._plumbing_request._events.put_noawait(self._plumbing_request.Event(None, None, None))
```

Comment:
I'm sure that one or more of those lines are redundant, what exactly do I need here?